### DISCUSSION TITLE
[21.11-d89f18a1] The Journey Continues!

### DISCUSSION THREAD

**Discussion by nzbr** [nix-community/NixOS-WSL#66] [Announcements]
This is the promised 21.11 release

It includes a change that allows for running code from foreign architectures by registering them through binfmt_misc, just like on native NixOS. This feature is enabled by default, but it may lead to WSLInterop (being able to run windows executables) breaking in other WSL distributions as a side effect. Should you experience issues with this, you can disable it by setting `wsl.interop.register = false` and removing all other binfmt registrations (if you have configured any). For more information on the underlying issue, see #64 and Microsoft/WSL#8203

Should you encounter any problems, feel free to [start a discussion](https://github.com/nix-community/NixOS-WSL/discussions/new) or open an issue!

### Flakes

Switch your nixpkgs ref to `github:NixOS/nixpkgs/nixos-21.11`  and run `nix flake update`

### Installer

Replace the content of `/etc/nixos/nixos-wsl` with the content of the source-tarball attached to this release

### Legacy (pre-installer)

Assuming that you followed the update procedure in the [last release](https://github.com/nix-community/NixOS-WSL/releases/tag/20.09-1c1f5649), replace `refs/tags/20.09-1c1f5649` with `refs/tags/21.11-d89f18a1`

# Changelog

## 🛠️ General Changes

* NixOS 21.11 (#65) @nzbr

## 🪲 Bugfixes

* binfmt / interop fixes (#64) @K900

<hr /><em>This discussion was created from the release <a href='https://github.com/nix-community/NixOS-WSL/releases/tag/21.11-d89f18a1'>[21.11-d89f18a1] The Journey Continues!</a>.</em>

**Comment by nzbr** [comment]
This release contains a known issue, that causes systemd to fail to start on first launch. Please restart the distribution with either `wsl -t NixOS` or `wsl --shutdown`.
For more information, see https://github.com/nix-community/NixOS-WSL/issues/67

**Reply by nzbr** [reply]
The issue has been resolved and a fixed binary has been attached to the release

**Reply by nabaco** [reply]
Any chance to add hashes of the files (MD5 or SHA)?

**Reply by nzbr** [reply]
Done!
